### ISSUE TITLE
Easy model build without dependencies that require sudo acces

### ISSUE THREAD

**Issue by martinberger** [riscv/sail-riscv#215]
In the discussion #212 on v0.15 planning, @allenjbaum stated that *"easy model build without dependencies that require sudo access"* is one of the two most important issues facing Sail at the moment.  This thread is meant as a place where we can bullet-point the core requirements for this. With a clear specification it is be easier to conceive of a solution.

**Comment by abukharmeh** [comment]
I think it could be really helpful if we make the CI upload the built model so people who are not intending to develop the the model can just download it and use it

**Comment by martinberger** [comment]
> upload the built model so ...

Isn't that trivial?

**Comment by abukharmeh** [comment]
Yep, just make the build directory job artifact

**Comment by Timmmm** [comment]
I agree, I think uploading the generated C would be a good first move. Then all you need is a C compiler and libgmp which is not that hard to install.

This is how we work around the pain of installing the Sail compiler internally.

Alternatively (or ideally additionally), it would be nice if there was a binary distribution of Sail (including Z3!) that you could just download instead of having to deal with 🤬ing OPAM.

**Comment by jordancarlin** [comment]
Seems like this can be closed now that we have a flow to distribute precompiled binaries (#715).

**Comment by Timmmm** [comment]
Yes I think so, that and the Sail binary release. On Linux at least! Mac should come with the next Sail release and I'll do Windows when someone asks for it...
